### PR TITLE
[FIRRTL][CAPI] Add the rest of setters for options

### PR DESCRIPTION
**Pull request by SpriteOvO**
This PR is based on ~PR #6435~ the `main` branch. Appreciate @darthscsi's work on rewriting the part of options for Firtool-bin.

Based on the description in that PR

> Interested parties are encouraged to finish out the set of functions.

So I'm trying to finish the rest work.

I proposed two solutions, but @darthscsi was probably too busy to comment any further on this. So I implemented them both as 2 PRs to further preview how they look and then we can compare and discuss which one is better.

---

This is the first solution, which manually implements the setter functions for all options without using macros. The second solution is #6439.

This PR also added a simple test for Firtool-lib C-API.

---

@darthscsi Would you mind making some constructive comments on this? And I'd love to hear if @mikeurbach and @seldridge have any opinions on this as well. Thanks!

### CODE REVIEW

**Comment by sequencer** [comment]
Discussed with @darthscsi please continue work on this PR.

**Comment by SpriteOvO** [comment]
PR rebased to `main` branch. @darthscsi Could you please review this PR? Waiting for your approval :)

**Review by seldridge — APPROVED** [review]
This aligns with the plan for exposing `Firtool.h` via the C-API.

I didn't go through this line-by-line.

File: include/circt-c/Firtool/Firtool.h
Comment (darthscsi) [reply include/circt-c/Firtool/Firtool.h:21]: yes, representing the namespace circt::firtool::firtooloptions

File: lib/CAPI/Firtool/Firtool.cpp
Comment (darthscsi) [inline lib/CAPI/Firtool/Firtool.cpp:19]: please keep the duplicate firtool.

**Review by darthscsi — APPROVED** [review]
LGTM, module minor nits.

**Comment by SpriteOvO** [comment]
Struct name has been changed back to `CirctFirtoolFirtoolOptions`, thanks for the reviews from @seldridge and @darthscsi!

**Pull request merged**
